File: backend/app/schemas/fixture_import.py
from pydantic import BaseModel


class FixtureImportRowPreview(BaseModel):
    row_number: int

    torneo: str | None = None
    anio: int | None = None
    fase: str | None = None
    grupo: str | None = None

    local: str | None = None
    codigo_local: str | None = None
    visitante: str | None = None
    codigo_visitante: str | None = None

    fecha_partido: str | None = None
    cierre_prediccion: str | None = None

    valid: bool
    errors: list[str] = []


class FixtureImportPreviewResponse(BaseModel):
    ok: bool
    total_rows: int
    valid_rows: int
    invalid_rows: int
    rows: list[FixtureImportRowPreview]


class FixtureImportCommitResponse(BaseModel):
    ok: bool
    total_rows: int
    imported_matches: int
    created_teams: int
    skipped_rows: int
    errors: list[str] = []

File: frontend/src/types/fixtureImport.ts
export interface FixtureImportRowPreview {
  row_number: number;

  torneo: string | null;
  anio: number | null;
  fase: string | null;
  grupo: string | null;

  local: string | null;
  codigo_local: string | null;
  visitante: string | null;
  codigo_visitante: string | null;

  fecha_partido: string | null;
  cierre_prediccion: string | null;

  valid: boolean;
  errors: string[];
}

export interface FixtureImportPreviewResponse {
  ok: boolean;
  total_rows: number;
  valid_rows: number;
  invalid_rows: number;
  rows: FixtureImportRowPreview[];
}

export interface FixtureImportCommitResponse {
  ok: boolean;
  total_rows: number;
  imported_matches: number;
  created_teams: number;
  skipped_rows: number;
  errors: string[];
}